### ISSUE TITLE
Add new networks: Goerli

### ISSUE THREAD

**Issue by anxolin** [cowprotocol/cowswap#416]
**Description**
Rinkeby will be deprecated soon. 

This issue is to add Goerli as a replacement.

I would leave Rinkeby for now, and coordinate with the backend the support for Goerli.

This task is perfect to also check what is needed to add Pollyon and other networks. The one doing it could create a document with the concise process to add a new Network. This will be handy when we add more networks. ideally, if we can make the process as simple as possible (maybe a refactor is needed)
https://drive.google.com/drive/folders/1NVfjnrNQ0e7WIEh-HM7N3lFFoVBqxJiU

Related issue
https://github.com/cowprotocol/services/issues/155

## Steps
- [x] Add Goerli network
- [ ] Coordinate version with backend https://github.com/cowprotocol/services/issues/155
     - blocked by API deployment with devops

Ideally, we would also do:
- [x] Document how to add new network
- [x] Refactor, to make it easier to add a new network (won't do for now)
- [ ] (opt) Ideally we would try to add as many networks as possible. We could bring at this point back the suggestion to add Polygon, and other networks.

## Context
https://cowservices.slack.com/archives/C0361CDD1FZ/p1650265111252219

**Comment by anxolin** [comment]
@nenadV91, I assigned you, but try to self assign yourself to the issues you are picking
